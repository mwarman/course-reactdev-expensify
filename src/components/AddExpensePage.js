import React from 'react';

const AddExpensePage = () => (
  <div>Add an Expense</div>
);

export default AddExpensePage;
