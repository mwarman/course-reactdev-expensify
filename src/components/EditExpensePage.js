import React from 'react';

const EditExpensePage = () => (
  <div>Edit an Expense</div>
);

export default EditExpensePage;
