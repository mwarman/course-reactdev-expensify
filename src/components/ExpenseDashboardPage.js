import React from 'react';

const ExpenseDashboardPage = () => (
  <div>Welcome to the Dashboard</div>
);

export default ExpenseDashboardPage;
