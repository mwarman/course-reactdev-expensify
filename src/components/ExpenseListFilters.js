import React from 'react';
import { connect } from 'react-redux';
// import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
// import 'react-dates/lib/css/_datepicker.css';

import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {

  state = {
    calendarFocused: null
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onCalendarFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  };

  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              placeholder="Search expenses"
              className="text-input"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            <select
              className="select"
              value={this.props.filters.sortBy}
              onChange={this.onSortChange}
            >
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDateId="filterStartDate"
              startDate={this.props.filters.startDate}
              endDateId="filterEndDate"
              endDate={this.props.filters.endDate}
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onCalendarFocusChange}
              showClearDates={true}
              numberOfMonths={1}
              isOutsideRange={() => false}
            />
          </div>
        </div>
      </div>
    );
  }

};

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (date) => dispatch(setStartDate(date)),
  setEndDate: (date) => dispatch(setEndDate(date))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
