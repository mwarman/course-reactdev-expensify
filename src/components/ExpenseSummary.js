import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => {
  const expenseWord = props.count === 1 ? 'expense' : 'expenses';

  return (
    <div>
      <h1>Viewing {props.count} {expenseWord} totalling {numeral(props.total / 100).format('$0,0.00')}.</h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    count: visibleExpenses.length,
    total: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
