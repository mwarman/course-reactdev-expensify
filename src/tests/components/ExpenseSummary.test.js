import React from 'react';
import { shallow } from 'enzyme';

import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should render ExpenseSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary count={2} total={10001} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with one expense', () => {
  const wrapper = shallow(<ExpenseSummary count={1} total={7599} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with no expenses', () => {
  const wrapper = shallow(<ExpenseSummary count={0} total={0} />);
  expect(wrapper).toMatchSnapshot();
});
