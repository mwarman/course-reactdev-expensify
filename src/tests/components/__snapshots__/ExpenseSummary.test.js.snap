// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $100.01
    .
  </h1>
</div>
`;

exports[`should render ExpenseSummary with no expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
    .
  </h1>
</div>
`;

exports[`should render ExpenseSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $75.99
    .
  </h1>
</div>
`;
